packages/legend-query-builder/src/components/QueryBuilderConstantExpressionPanel.tsx

```diff
@@ -86,6 +89,8 @@ const QueryBuilderConstantExpressionEditor = observer(
         );
     const validationMessage = !variableName
       ? `Constant Name can't be empty`
+      : validateValidString(variableName) === false
```

Comment:
the one thing we need to add here is we must block the saving, right now we only show the warning, but users are still able to create/edit the variable

Reply:
makes sense!